### PR TITLE
Use ale instead of syntastic

### PR DESCRIPTION
**Pull request by ryanhinkel**
Ale lints and fixes code asychronously so the editor is not blocked while the file saves. Its great.

### CODE REVIEW

**Comment by patricksmithdds** [comment]
@ryanhinkel Do we want to include the default ALE configuration in this branch?

**Comment by patricksmith** [comment]
@ryanhinkel Do we want to include the default ALE configuration in this branch?

**Comment by rzane** [comment]
You might want to include a little bit of configuration. For example, here's what I have in my `vimrc`: https://github.com/rzane/dotfiles/blob/eaff1e6ea0eae71818d3209564130bf5d6710e76/vim/vimrc#L348-L373.

File: vimrc [hunk 1]
```diff
@@ -359,3 +359,41 @@ endif
 
 " ensure vim always runs from a shell
 set shell=/bin/sh
+
+
+""""""""""""""""""""""
```
Comment (rzane) [inline vimrc]: @ryanhinkel, it's  important that this configuration comes *before* loading `vimrc.local` otherwise, people can't override these settings.

**Comment by ryanhinkel** [comment]
Good catch @rzane

**Pull request merged**